src/ansys/mechanical/core/embedding/logger/environ.py

```diff
@@ -0,0 +1,104 @@
+        else:
+            # only disable global logging if none of the sinks match?
+            os.environ["ANSYS_WORKBENCH_LOGGING"] = "0"
+
+    def set_log_level(self, level: int, sink: int = sinks.StandardSinks.CONSOLE):
+        """Set the log level for the Mechanical application based on the python log level."""
+        if level == logging.NOTSET:
+            self.disable(sink)
+            return
+
+        if level <= logging.DEBUG:
+            # level 0 in workbench logging is trace, level 1 is debug.
+            # python logging.DEBUG will imply trace.
+            os.environ["ANSYS_WORKBENCH_LOGGING_FILTER_LEVEL"] = "0"
+        elif level <= logging.INFO:
+            os.environ["ANSYS_WORKBENCH_LOGGING_FILTER_LEVEL"] = "2"
+        elif level <= logging.WARNING:
+            os.environ["ANSYS_WORKBENCH_LOGGING_FILTER_LEVEL"] = "3"
+        elif level <= logging.ERROR:
+            os.environ["ANSYS_WORKBENCH_LOGGING_FILTER_LEVEL"] = "4"
+        elif level <= logging.CRITICAL:
+            os.environ["ANSYS_WORKBENCH_LOGGING_FILTER_LEVEL"] = "5"
+
+    def set_auto_flush(self, flush: bool) -> None:
+        """Set whether to auto flush to the standard log file."""
+        value = "1" if flush else "0"
+        os.environ["ANSYS_WORKBENCH_LOGGING_AUTO_FLUSH"] = value
+
+    def set_directory(self, directory: str) -> None:
+        """Set the location to write the log file to."""
+        os.environ["ANSYS_WORKBENCH_LOGGING_DIRECTORY"] = directory
+
+    def set_base_directory(self, base_directory: str) -> None:
+        """Set the base location to write the log file to.
+
+        The base directory contains time-stamped subfolders where the log file
+        is actually written to. This takes precedence over set_directory if set
+        """
+        os.environ["ANSYS_WORKBENCH_LOGGING_BASE_DIRECTORY"] = base_directory
+
+    def can_log_message(self, level: int) -> bool:
+        """Return whether a message with the given severity will be output into the log."""
+        if os.environ.get("ANSYS_WORKBENCH_LOGGING", 0) == 0:
+            return False
+
+        wb_int_level = int(os.environ.get("ANSYS_WORKBENCH_LOGGING_FILTER_LEVEL", 2))
+        if wb_int_level == 0:
+            return True
+        if wb_int_level == 1:
+            # This is not exactly right. WB might be set to DEBUG but the level expected is trace.
+            return True
+        if wb_int_level == 2:
+            return level >= logging.INFO
+        if wb_int_level == 3:
+            return level >= logging.WARNING
+        if wb_int_level == 4:
+            return level >= logging.ERROR
+        if wb_int_level == 5:
+            return level >= logging.CRITICAL
+
+    def log_message(self, level: int, context: str, message: str) -> None:
+        """Log the message to the configured logging mechanism."""
+        raise Exception(
+            "Can't log to the embedding logger until the Mechanical application is initialized"
```

Comment:
```suggestion
            "Can't log to the embedding logger until Mechanical is initialized."
```

Reply:
@PipKat no reason, thanks for looking over this with such a fine-tooth comb